src/main/java/io/confluent/kafkarest/entities/TopicPartitionOffsetMetadata.java

```diff
@@ -97,7 +97,7 @@ public boolean equals(Object o) {
     if (offset != that.offset) {
       return false;
     }
-    if (metadata != that.metadata) {
```

Comment:
I feel you might be mistaken about `Objects.equals()`.  Here's how it's implemented on my JDK:

```java
    /**
     * Returns {@code true} if the arguments are equal to each other
     * and {@code false} otherwise.
     * Consequently, if both arguments are {@code null}, {@code true}
     * is returned and if exactly one argument is {@code null}, {@code
     * false} is returned.  Otherwise, equality is determined by using
     * the {@link Object#equals equals} method of the first
     * argument.
     *
     * @param a an object
     * @param b an object to be compared with {@code a} for equality
     * @return {@code true} if the arguments are equal to each other
     * and {@code false} otherwise
     * @see Object#equals(Object)
     */
    public static boolean equals(Object a, Object b) {
        return (a == b) || (a != null && a.equals(b));
    }

```

More info at https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#equals(java.lang.Object)

In other words, it does a reference check as a shortcut (also conveniently works for `null`s), and then does a `.equals()` as one would expect.

I feel that expresses the intention much cleaner and more succinctly than what is done for `topic`, and it's a Java 7 API (as opposed to Java 8) so it preserves Java 7 compatibility.